mitxgraders/formulagrader/intervalgrader.py

```diff
@@ -0,0 +1,231 @@
+"""
+intervalgrader.py
+
+Grader for intervals consisting of two formula entries, as well
+as enclosing brackets.
+"""
+from __future__ import print_function, division, absolute_import, unicode_literals
+
+from voluptuous import Required, Schema, Any, All, Length
+
+from mitxgraders.baseclasses import AbstractGrader
+from mitxgraders.stringgrader import StringGrader
+from mitxgraders.listgrader import SingleListGrader, demand_no_empty
+from mitxgraders.formulagrader.formulagrader import FormulaGrader, NumericalGrader
+from mitxgraders.exceptions import ConfigError, InvalidInput, MissingInput
+from mitxgraders.helpers.validatorfuncs import text_string
+
+class IntervalGrader(SingleListGrader):
+    """
+    IntervalGrader grades a pair of numbers describing an interval.
+    Both the interval bracketing and the numbers are graded.
+    The following are examples of possible entries:
+    * [1, 2)
+    * {0, 1+pi]
```

Comment:
The `{` isn't a typo, is it? You're pointing out that `{` is possible with configuration?

(Incidentally, `{` seems useful for discrete sets, but in that scenario we wouldn't want the two-element limitation. I suppose that is out-of-scope for the current work, though)

Reply:
I'll change it to a default setting, and add that configuration options will allow other brackets (eg, `{}` and `<>`).